tests/test_isrTask.py

```diff
@@ -0,0 +1,529 @@
+        self.task.updateVariance(self.inputExp, self.amp)
+        statAfter = computeImageMedianAndStd(self.inputExp.variance[self.amp.getBBox()])
+        self.assertGreater(statAfter[0], statBefore[0])
+
+    def test_darkCorrection(self):
+        """Expect the median image value should decrease after this operation.
+        """
+        darkIm = isrMock.DarkMock().run()
+
+        statBefore = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.task.darkCorrection(self.inputExp, darkIm)
+        statAfter = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.assertLess(statAfter[0], statBefore[0])
+
+    def test_darkCorrection_noVisitInfo(self):
+        """Expect the median image value should decrease after this operation.
+        """
+        darkIm = isrMock.DarkMock().run()
+        darkIm.getInfo().setVisitInfo(None)
+
+        statBefore = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.task.darkCorrection(self.inputExp, darkIm)
+        statAfter = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.assertLess(statAfter[0], statBefore[0])
+
+    def test_flatCorrection(self):
+        """Expect the image median should increase (divide by < 1).
+        """
+        flatIm = isrMock.FlatMock().run()
+
+        statBefore = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.task.flatCorrection(self.inputExp, flatIm)
+        statAfter = computeImageMedianAndStd(self.inputExp.image[self.amp.getBBox()])
+        self.assertGreater(statAfter[1], statBefore[1])
+
+    def test_saturationDetection(self):
+        """Expect the saturation level detection/masking to scale with
+        threshold.
+        """
+        self.amp.setSaturation(1000.0)
+        self.task.saturationDetection(self.inputExp, self.amp)
+        countBefore = countMaskedPixels(self.mi, "SAT")
+
+        self.amp.setSaturation(25.0)
+        self.task.saturationDetection(self.inputExp, self.amp)
+        countAfter = countMaskedPixels(self.mi, "SAT")
+
+        self.assertLessEqual(countBefore, countAfter)
+
+    def test_measureBackground(self):
+        """Expect the background measurement runs successfully and to save
+        metadata values.
+        """
+        self.config.qa.flatness.meshX = 20
+        self.config.qa.flatness.meshY = 20
+        self.task.measureBackground(self.inputExp, self.config.qa)
+        self.assertIsNotNone(self.inputExp.getMetadata().getScalar('SKYLEVEL'))
+
+    def test_flatContext(self):
+        """Expect the flat context manager runs successfully and leaves the
+        image data the same.
+        """
+        darkExp = isrMock.DarkMock().run()
+        flatExp = isrMock.FlatMock().run()
+
+        mi = self.inputExp.getMaskedImage().clone()
+        with self.task.flatContext(self.inputExp, flatExp, darkExp):
+            self.assertTrue(True)
```

Comment:
What is this doing?

Reply:
Nothing (but the goal is to test that the flat context manager doesn't alter the image), and so is now replaced with pass.

Reply:
Wouldn't the test then be `self.assertEqual(self.inputExp, flatContext(self.inputExp)` or something like that?

Reply:
I don't think so.  The flatContext() should be applying the flat and dark to the input exposure, passing (in the current test) instead of operating on that now-processed exposure, and then removing the flat and dark so the input exposure can be operated on in the "clean" state.  The goal here is to test that this context based round-trip leaves the exposure the same as before.

Reply:
Oh, I see: you're making that test just below (it was confusing without the test docstring). Is there any way to identify that inside the context manager the image has been modified correctly?

Reply:
As the context manager calls the same flatCorrection/darkCorrection methods, I was hoping the tests on those methods would cover that code.  I have added value tests on the statistics to do ensure that unexpected code changes trigger an error.